lib/mongo/protocol/msg.rb

```diff
@@ -275,6 +275,28 @@ def bulk_write?
         num_inserts > 1  || num_updates > 1 || num_deletes > 1
       end
 
+      def maybe_add_server_api(client)
+        if client.nil?
+          raise ArgumentError, 'Client must be provided'
+        end
+
+        if server_api = client.options[:server_api]
+          main_document = @main_document.dup
+          if version = server_api[:version]
+            main_document['apiVersion'] = version
+          end
+          if server_api[:strict]
```

Comment:
Seeing this check made me realise that the spec may need some extra tests to cover this bit of the spec:
> Drivers MUST add the `apiStrict` and `apiDeprecationErrors` options if they were specified by the user, even when the specified value is equal to the server default.

If the `strict` option was not specified by the user it should be omitted, but if it was specified than it should indeed be sent to the server, even when it's `false`.

Reply:
Changed.